### PR TITLE
Progress logs for templated pages

### PR DESCRIPTION
**Pull request by phukon**
### Description
#1852 

Implement feature as a Vite plugin to count markdown files in the pages directory and compare them with the number of pages built during the Vite build process. This will provide the build progression percentage specifically for templated pages.

### CODE REVIEW

**Comment by mcrascal** [comment]
Epic, thank you!

File: packages/lib/plugin-connector/src/build-plugins/verbose-logs/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/** @type {import("vite").Plugin} */
+export const verboseLogs = {
+	name: 'evidence:verbose-logs',
+
+	buildStart() {
+		const directoryPath = '.svelte-kit/output/prerendered/pages';
+		getPagesDir('../../pages')
+			.then((totalExpectedFiles) => {
+				// console.log('Build started');
```
Comment (archiewood) [inline packages/lib/plugin-connector/src/build-plugins/verbose-logs/index.js:12]: A few of these hanging around to be removed

**Review by archiewood — COMMENTED** [review]
Sorely needed feature!

What's the current UX for using this plugin with `npm run build`

**Comment by phukon** [comment]
### New approach [wip]
Goal: Forked childprocess tracks the progess on each template file (route)


https://github.com/evidence-dev/evidence/assets/60285613/ac60935b-92e0-408a-a688-25d3626f0512

**Comment by archiewood** [comment]
> Should I progress further or is it complicating a very trivial thing?

My instinct is this is a lot of additional LoC and some more deps for a useful, but not _essential_ feature.

Is there a simpler approach we can run using something like fs or [chokidar](https://github.com/paulmillr/chokidar) - existing dep

**Comment by phukon** [comment]
Minimal

https://github.com/evidence-dev/evidence/assets/60285613/3fe879a8-f77c-4894-b039-6235d5223960

**Comment by phukon** [comment]
Minimal

https://github.com/evidence-dev/evidence/assets/60285613/3fe879a8-f77c-4894-b039-6235d5223960

**Comment by phukon** [comment]
Minimal

https://github.com/evidence-dev/evidence/assets/60285613/3fe879a8-f77c-4894-b039-6235d5223960

**Comment by ItsMeBrianD** [comment]
Big fan of this feature - overall this PR looks good, but I don't think the plugin connector is the best place to land it

**Comment by ItsMeBrianD** [comment]
I've logged this as an issue in #2258, I'm going to close this until it can be re-addressed in that issue, but this is a great feature and we appreciate the submission!

**Pull request closed without merging**